units/provisioning_convert_host_to_legacy.go

```diff
@@ -0,0 +1,225 @@
+	j := &convertHostToLegacyProvisioningJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    convertHostToLegacyProvisioningJobName,
+				Version: 0,
+			},
+		},
+	}
+	j.UpdateTimeInfo(amboy.JobTimeInfo{
+		MaxTime: host.MaxLCTInterval,
+	})
+	j.SetDependency(dependency.NewAlways())
+	return j
+}
+
+// NewConvertHostToLegacyProvisioningJob converts a host from a legacy provisioned
+// host to a non-legacy provisioned host.
+func NewConvertHostToLegacyProvisioningJob(env evergreen.Environment, h host.Host, id string, attempt int) amboy.Job {
+	j := makeConvertHostToLegacyProvisioningJob()
+	j.env = env
+	j.host = &h
+	j.HostID = h.Id
+	j.CurrentAttempt = attempt
+	j.SetPriority(1)
+	j.SetID(fmt.Sprintf("%s.%s.%s", convertHostToLegacyProvisioningJobName, j.HostID, id))
+	return j
+}
+
+func (j *convertHostToLegacyProvisioningJob) Run(ctx context.Context) {
+	defer j.MarkComplete()
+
+	if err := j.populateIfUnset(); err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "could not populate required fields",
+			"host":    j.HostID,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+		j.AddError(err)
+		return
+	}
+
+	if j.host.NeedsReprovision != host.ReprovisionToLegacy || j.host.Status != evergreen.HostProvisioning {
+		return
+	}
+
+	defer func() {
+		if j.HasErrors() {
+			event.LogHostConvertingProvisioningError(j.host.Id, j.Error())
+
+			if err := j.tryRequeue(ctx); err != nil {
+				grip.Error(message.WrapError(err, message.Fields{
+					"message": "could not enqueue job to retry provisioning conversion",
+					"host":    j.host.Id,
+					"distro":  j.host.Distro.Id,
+					"job":     j.ID(),
+				}))
+				j.AddError(err)
+			}
+			grip.Error(message.WrapError(j.Error(), message.Fields{
+				"message": "failed to convert host to legacy provisioning but will retry",
+				"host":    j.host.Id,
+				"distro":  j.host.Distro.Id,
+				"job":     j.ID(),
+			}))
+		}
+		grip.Error(message.WrapError(j.host.SetReprovisioningLocked(false), message.Fields{
+			"message": "could not clear host reprovisioning lock",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+	}()
+
+	// Lock the provisioning state to prevent other provisioning jobs from
+	// running.
+	if err := j.host.SetReprovisioningLockedAtomically(true); err != nil {
+		grip.Info(message.WrapError(err, message.Fields{
+			"message": "reprovisioning job currently in progress, returning from job",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+		return
+	}
+
+	if err := j.host.UpdateLastCommunicated(); err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "could not update host communication time",
+			"host":    j.host.Id,
+			"distro":  j.host.Distro.Id,
+			"job":     j.ID(),
+		}))
+	}
+
+	// The host cannot be reprovisioned until the host's agent monitor has
+	// stopped.
+	if !j.host.NeedsNewAgent {
```

Comment:
Check this at the top so that we don't run other unnecessary checks. Also does this need to requeue? The defer won't requeue if we hit this

Reply:
Done. I requeued if this block executes.